### PR TITLE
add Pygments lexer for Singularity definition files

### PR DESCRIPTION
**Pull request by hall**
## Description of the Pull Request (PR):

These changes add a local Pygments lexer for Singularity definition files; e.g.:

deepinscreenshot_select-area_20181230010709

Here's a test file (an image since GitHub doesn't allow HTML):

deepinscreenshot_select-area_20181230010603

I'll also be adding this into the upstream repo so feel free to close this PR if you'd rather wait for that. I'm not incredibly familiar with the definition file syntax so I'm mostly looking for feedback before attempting to add the lexer into Pygments.

## This fixes or addresses the following GitHub issues:

- None

### CODE REVIEW

**Review by jmstover — APPROVED** [review]
This looks good to me.

@devprincess ?? You have much more experience here with Sphinx, etc.. than I.

File: pygments_singularity.py [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+    <https://www.sylabs.io/guides/3.0/user-guide/definition_files.html>`_.
+    """
+
+    name = 'Singularity'
+    aliases = ['singularity']
+    filenames = ['*.def', 'Singularity']
+    flags = re.IGNORECASE | re.MULTILINE | re.DOTALL
+
+    _headers = r'^(\s)*(bootstrap|from|osversion|mirrorurl|include)(:)'
+    _section = r'^%(?:post|setup|environment|help|labels|test|runscript|files|startscript)\b'
```
Comment (GodloveD) [inline pygments_singularity.py]: Add `pre` to `_section` (undocumented as of now).

File: pygments_singularity.py [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+    Lexer for `Singularity definition files
+    <https://www.sylabs.io/guides/3.0/user-guide/definition_files.html>`_.
+    """
+
+    name = 'Singularity'
+    aliases = ['singularity']
+    filenames = ['*.def', 'Singularity']
+    flags = re.IGNORECASE | re.MULTILINE | re.DOTALL
+
+    _headers = r'^(\s)*(bootstrap|from|osversion|mirrorurl|include)(:)'
```
Comment (GodloveD) [inline pygments_singularity.py]: Some other `_headers` should be added.  For instance, `Registry`, `Namespace`, `IncludeCMD`.  See the `Keywords` sections in the appendix for info.

**Review by GodloveD — CHANGES_REQUESTED** [review]
@hall thanks very much for this contribution!  I think it will really help make the docs more readable, and the syntax highlighting is a useful addition to the pygments package as well.

Just a few suggestions for improvement here. 

I'm happy to include this as is.  Maybe you can submit another PR when this is merged upstream?  

Thanks again!

**Comment by hall** [comment]
Thanks for the feedback, @GodloveD! I'll make a note to remove these changes after they're merged upstream.

**Comment by hall** [comment]
Here's the upstream PR:

https://bitbucket.org/birkenfeld/pygments-main/pull-requests/793/add-singularity-lexer/diff

**Review by GodloveD — APPROVED** [review]

**Comment by GodloveD** [comment]
Thanks again @hall !

**Pull request merged**